### PR TITLE
DOC: Fix typos in NEPs, found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
As suggested in https://github.com/numpy/numpy/pull/19911#issuecomment-924436349.

### CODE REVIEW

File: doc/neps/nep-0012-missing-data.rst [hunk 1]
```diff
@@ -903,7 +903,7 @@ before it will allow NA-masked arrays to flow through.
 https://docs.scipy.org/doc/numpy/reference/c-api.array.html#NPY_ARRAY_ALLOWNA
 
 Code which does not follow this advice, and instead just calls PyArray_Check() to verify
-its an ndarray and checks some flags, will silently produce incorrect results. This style
+it's an ndarray and checks some flags, will silently produce incorrect results. This style
```
Comment (seberg) [inline doc/neps/nep-0012-missing-data.rst:906]: Personally, I would prefer spelling the "is" out.  Formal written out text should not normally use `'`?  The usable surprised me, but seems like generally the more typical spelling.

Comment (DimitriPapadopoulos) [reply doc/neps/nep-0012-missing-data.rst:906]: Yes, _useable_ seems correct in American English, although you will find tenfold more _usable_ hits in Google. British dictionaries tend to mention only _usable_.

Comment (DimitriPapadopoulos) [reply doc/neps/nep-0012-missing-data.rst]: Changed `it's` to `it is`.

Comment (charris) [reply doc/neps/nep-0012-missing-data.rst]: > British dictionaries tend to mention only _usable_.

"England and America are two countries separated by a common language" -- GBS :)

**Comment by seberg** [comment]
Aside from the `its` (and the `us(e)able`), all of these seem like clear typos so lets put this in.  Thanks.

**Pull request merged**